services/web/server/src/simcore_service_webserver/projects/_crud_api_read.py

```diff
@@ -143,6 +142,28 @@ async def list_projects(  # pylint: disable=too-many-arguments
     return projects, total_number_projects
 
 
+async def list_projects_full_search(
+    app,
+    *,
+    user_id: UserID,
+    product_name: str,
+    offset: NonNegativeInt,
+    limit: int,
+    text: str | None,
```

Comment:
See my comment on adding some structure to the search below

Reply:
The same as below.